implementations/rust/ockam/ockam_api/src/enroll/enrollment.rs

```diff
@@ -110,14 +118,31 @@ impl Enrollment for SecureClient {
     }
 
     #[instrument(skip_all)]
-    async fn present_token(&self, ctx: &Context, token: &OneTimeCode) -> miette::Result<()> {
+    async fn present_token(
+        &self,
+        ctx: &Context,
+        token: &OneTimeCode,
+    ) -> miette::Result<EnrollStatus> {
         let req = Request::post("/").body(token);
         trace!(target: TARGET, "present a token");
-        self.tell(ctx, DefaultAddress::ENROLLMENT_TOKEN_ACCEPTOR, req)
+        match self
+            .tell(ctx, DefaultAddress::ENROLLMENT_TOKEN_ACCEPTOR, req)
             .await
             .into_diagnostic()?
-            .success()
-            .into_diagnostic()
+        {
+            Reply::Successful(_) => Ok(EnrollStatus::EnrolledSuccessfully),
+            Reply::Failed(e, s) => match (e.message(), s) {
+                // TODO: the `authenticator` should return proper error codes
```

Comment:
The authenticator could return a `201` when adding it, and a `200` when it already exists

Reply:
Yeah, I briefly talked about this with @SanjoDeundiak. The main problem is that it requires a big refactor to change how results/errors are handled in that module.